### PR TITLE
Add support for Get/Update API Shield Schema Validation Settings

### PR DESCRIPTION
**Pull request by djhworld**
This change adds support for the following API Shield related endpoints related to API Shield Schema Validation Settings:

- Retrieve API Shield Schema Validation Settings https://developers.cloudflare.com/api/operations/api-shield-schema-validation-retrieve-zone-level-settings
- Update All API Shield Schema Validation Settings https://developers.cloudflare.com/api/operations/api-shield-schema-validation-update-zone-level-settings
- Update API Shield Schema Validation Settings https://developers.cloudflare.com/api/operations/api-shield-schema-validation-patch-zone-level-settings

## Has your change been tested?

* Unit tests
* Local testing

## Types of changes

What sort of change does your code introduce/modify?

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

**Comment by djhworld** [comment]
Draft PR, awaiting API documentation

File: api_shield_schemas.go [hunk 1]
```diff
@@ -333,3 +333,114 @@ func (api *API) UpdateAPIShieldSchema(ctx context.Context, rc *ResourceContainer
...
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+
+	return &asResponse.Result, nil
+}
+
+// UpdateAllAPIShieldSchemaValidationSettings updates zone level schema validation settings
+//
+// API documentation: https://developers.cloudflare.com/api/operations/api-shield-schema-validation-update-zone-level-settings
+func (api *API) UpdateAllAPIShieldSchemaValidationSettings(ctx context.Context, rc *ResourceContainer, params UpdateAllAPIShieldSchemaValidationSettingsParams) (*APIShieldSchemaValidationSettings, error) {
```
Comment (jacobbednarz) [inline api_shield_schemas.go]: the way we would generally handle this is that we wouldn't include this method if the individual fields update endpoint was available (such as `UpdateAPIShieldSchemaValidationSettings`). we would opt to instead use the one method to update a single or all of the fields. does that make sense to do in this context?

Comment (djhworld) [reply api_shield_schemas.go]: Makes sense! Have removed this in favour of the PATCH endpoint. 

https://github.com/cloudflare/cloudflare-go/compare/0d0a92a41e5c180aa57f6a06b77aec34c4b34764..c52383a53920cfc4d5c93170c38ddcfe5e3fc183

**Comment by jacobbednarz** [comment]
wicked effort! thanks @djhworld 👏

**Pull request merged**